Tools/CISupport/Shared/de-duper

```diff
@@ -114,21 +119,28 @@ def main(args):
             if os.stat(full_path).st_nlink > 1:
                 continue
             sha_value = sha1(full_path)
+            sha_path = os.path.join(args.storage, f'{sha_value}.png')
+            if sha_path == full_path:
+                continue
             if sha_value in existing_links:
                 if args.dry_run:
                     print(f'[dry-run] Linking {full_path} to {sha_value}.png')
                 else:
                     os.remove(full_path)
-                    os.link(os.path.join(args.storage, f'{sha_value}.png'), full_path)
+                    os.link(sha_path, full_path)
+                    existing_links[sha_value] += 1
+                    if existing_links[sha_value] > MAX_LINKS:
+                        os.remove(sha_path)
+                        del existing_links[sha_value]
```

Comment:
This behaviour seems quite subtle and probably worth commenting: we're effectively allowing a new file to be created in storage once we reach `MAX_LINKS`, but this doesn't impact the existing files linked to the file we're removing from this path.